### ISSUE TITLE
ICML '2020 | Generative Pretraining from Pixels

### ISSUE THREAD

**Issue by vycezhong** [vycezhong/read-papers#74]
https://cdn.openai.com/papers/Generative_Pretraining_from_Pixels_V2.pdf

from: Openai blog

**Comment by vycezhong** [comment]
这篇文章做的就是image gpt/bert的工作 #14 

直接把图像当作输入，会有一个问题：序列太长了，而transformer的self attention又是O(N^2)的复杂度. 所以需要减少序列长度. 第一步是resize,把224x224x3 resize成32x32x3，因为如果低于32人类判断也会不准确. 但这还是太长了. 他们又搞了一个色彩的encoding, 又缩短了3倍. 1024这个长度还算可以接受. 

剩下的就和train gpt/bert一样了. 他们是在IMAGENET上做pre-train 1) gpt式：预测下一个pixel 2) BERT式：MASK , 然后在CIFAR10 / CIFAR100 / STL-10上做 1) 直接linear prob 2) finetune. 


实验1 
![image](https://user-images.githubusercontent.com/25879526/87761997-f57a8480-c844-11ea-8b37-f135a08bec9f.png)
pretrain的模型，并不是最后一层最适合去做classification，反而是中间的层特征更好. 对于有监督训练来讲，一般都是倒数第二层的特征是最好的. 但是对于pretraining来讲，生成下一步pixel这个目标和分类是没有关系的，所以不见得倒数第二层的特征最好. 

实验2 
![image](https://user-images.githubusercontent.com/25879526/87762512-d7f9ea80-c845-11ea-878a-99d73097f45a.png)
pretrain的时候loss越低，在downstream task上的accuracy也越高

实验3
![image](https://user-images.githubusercontent.com/25879526/87763335-370c2f00-c847-11ea-80ff-096661c5ec70.png)
跟其他方法在CIFAR10/100上对比了下，略胜一筹 Sup是指有监督，UnSup是无监督.

实验4
![image](https://user-images.githubusercontent.com/25879526/87763483-72a6f900-c847-11ea-89c5-e672bd9786b5.png)
跟其他方法在IMAGENET上对比了下，结果很差啊. SimCLR明显吊打其他.

实验5
![image](https://user-images.githubusercontent.com/25879526/87763632-aeda5980-c847-11ea-9a5c-f1773189e82c.png)
加上finetune后的结果. 半斤八两.

实验6
![image](https://user-images.githubusercontent.com/25879526/87763811-f6f97c00-c847-11ea-927f-28af5d841dec.png)
(蓝色是直接linear prob, 橙色的是加上了finetune ，加粗橙色是ensembling bert masks)
这个实验比较了gpt式模型和bert式模型.. 如果只做linear prob,那bert模型被吊打. 但是加上了finetune后，半斤八两了. 

实验7
![image](https://user-images.githubusercontent.com/25879526/87765018-daf6da00-c849-11ea-9bc7-9f29e3ece2d5.png)
low-data的比较，不是很熟. 准确率好低啊.
